client/src/controllers/InitController.test.js

```diff
@@ -148,4 +148,39 @@ describe('InitController', () => {
       ]);
```

Comment:
We should also add a test in this file for the change in behaviour where the other data attributes will now be removed.

Further up in this test, here's a recommended place to add a single line to test that all other data attributes get removed when we 'clean up' the controller attributes.

```js
    it('should remove the controller correctly', () => {
      const testDiv = document.getElementById('test');

      expect(handleEvent).toHaveBeenCalledTimes(1);
      expect(testDiv.getAttribute('data-controller')).toBeNull();
      expect({ ...testDiv.dataset }).toEqual({}); # Add this line
    });
```